src/NUnitFramework/tests/Constraints/CollectionEquivalentConstraintTests.cs

```diff
@@ -152,8 +152,19 @@ public void EquivalentHonorsUsing()
                 .Using<string>((x, y) => StringUtil.Compare(x, y, true))
                 .ApplyTo(set2).IsSuccess);
         }
+#if (NET_4_0 || NET_4_5) && !PORTABLE
 
-#if ( NET_4_0 || NET_4_5 ) && !PORTABLE
+        [Test]
+        public void EquivalentHonorsUsingWhenCollectionsAreOfDifferentTypes()
+        {
+            ICollection set1 = new SimpleObjectCollection("1", "2", "3");
+            ICollection set2 = new SimpleObjectCollection(1, 2, 3);
+
+            Assert.That(set2, Is.EquivalentTo(set1).Using<int, string>((i, s) => i.ToString() == s));
+        }
+#endif
```

Comment:
I don't see this logic will make sense for UniqueItemsContraint. I'm probably going to just implement it in the other three.